front/lib/api/assistant/feedback.ts

```diff
@@ -0,0 +1,244 @@
+import type {
+  ConversationType,
+  ConversationWithoutContentType,
+  Result,
+} from "@dust-tt/types";
+import type { UserType } from "@dust-tt/types";
+import { ConversationError, Err, GLOBAL_AGENTS_SID, Ok } from "@dust-tt/types";
+import { Op } from "sequelize";
+
+import { canAccessConversation } from "@app/lib/api/assistant/conversation";
+import type { AgentMessageFeedbackDirection } from "@app/lib/api/assistant/conversation/feedbacks";
+import type { Authenticator } from "@app/lib/auth";
+import { AgentConfiguration } from "@app/lib/models/assistant/agent";
+import {
+  AgentMessage,
+  AgentMessageFeedback,
+} from "@app/lib/models/assistant/conversation";
+import { Message } from "@app/lib/models/assistant/conversation";
+import { AgentMessageFeedbackResource } from "@app/lib/resources/agent_message_feedback_resource";
+
+/**
+ * We retrieve the feedbacks for a whole conversation, not just a single message.
+ */
+
+export type AgentMessageFeedbackType = {
+  id: number;
+  messageId: string;
+  agentMessageId: number;
+  userId: number;
+  thumbDirection: AgentMessageFeedbackDirection;
+  content: string | null;
+};
+
+export async function getConversationFeedbacksForUser(
+  auth: Authenticator,
+  conversation: ConversationType | ConversationWithoutContentType
+): Promise<Result<AgentMessageFeedbackType[], ConversationError>> {
+  const owner = auth.workspace();
+  if (!owner) {
+    throw new Error("Unexpected `auth` without `workspace`.");
+  }
+  const user = auth.user();
+  if (!canAccessConversation(auth, conversation) || !user) {
+    return new Err(new ConversationError("conversation_access_restricted"));
+  }
+
+  const messages = await Message.findAll({
+    where: {
+      conversationId: conversation.id,
+      agentMessageId: {
+        [Op.ne]: null,
+      },
+    },
+    attributes: ["sId", "agentMessageId"],
+  });
+
+  const agentMessages = await AgentMessage.findAll({
+    where: {
+      id: {
+        [Op.in]: messages
+          .map((m) => m.agentMessageId)
+          .filter((id): id is number => id !== null),
+      },
+    },
+  });
+
+  const feedbacks = await AgentMessageFeedback.findAll({
+    where: {
+      userId: user.id,
+      agentMessageId: {
+        [Op.in]: agentMessages.map((m) => m.id),
+      },
+    },
+  });
```

Comment:
We should use the resource as we have one

Reply:
Yup, my bad. Fixed